boot/initramfs.go

```diff
@@ -125,3 +127,61 @@ func MockInitramfsReboot(f func() error) (restore func()) {
 func InitramfsReboot() error {
 	return initramfsReboot()
 }
+
+// This function implements logic that is usually part of the
+// bootloader, but that it is not possible to implement in, for
+// instance, piboot. See handling of kernel_status in
+// bootloader/assets/data/grub.cfg.
+func updateNotScriptableBootloaderStatus(bl bootloader.NotScriptableBootloader) error {
+	blVars, err := bl.GetBootVars("kernel_status")
+	if err != nil {
+		return err
+	}
+	curKernStatus := blVars["kernel_status"]
+	if curKernStatus == "" {
+		return nil
+	}
+
+	cmdLine, err := osutil.KernelCommandLine()
+	if err != nil {
+		return err
+	}
+	args := strings.Split(cmdLine, " ")
+	// "" would be the value for the error case
```

Comment:
Can you clarify this better? By reading the code, it seems that `newStatus` will be empty in a few cases:
1. If  `kernel_status == trying` but `curKernStatus != try` (and this would look to me like an error/impossible situation, am I correct?)
2. If `kernel_status != trying` but `curKernStatus == try` (it means that the RPi was powered off after the update instead of being rebooted, right?)
3. If `kernel_status != trying` and `curKernStatus != try` (this doesn't look like an error situation, but rather that the previous update went smoothly?)

Reply:
1. Yes, this case should be impossible
2. It can mean also a failed kernel update (that's a problem that we have in piboot, we cannot differentiate between failed updates and power off before the reboot happens)
3. Note that for piboot, when `updateNotScriptableBootloaderStatus()` is called `curKernStatus` can be only `try` or `""`, and for the second case you have already returned a few lines before. Any other value different from "try" at this line is therefore an error.

I've added more information in the comment to clarify this.